atom/browser/api/atom_api_update_client.cc

```diff
@@ -0,0 +1,172 @@
+// Copyright 2016 The Brave Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "atom/browser/api/atom_api_update_client.h"
+
+#include "atom/browser/api/atom_api_extension.h"
+#include "atom/browser/component_updater/default_extensions.h"
+#include "atom/browser/component_updater/extension_installer.h"
+#include "atom/browser/extensions/atom_extension_system.h"
+#include "atom/browser/extensions/atom_notification_types.h"
+#include "atom/common/native_mate_converters/string16_converter.h"
+#include "atom/common/native_mate_converters/value_converter.h"
+#include "base/strings/utf_string_conversions.h"
+#include "base/values.h"
+#include "brave/browser/brave_content_browser_client.h"
+#include "components/component_updater/component_updater_service.h"
+#include "components/update_client/crx_update_item.h"
+#include "content/public/browser/browser_thread.h"
+#include "content/public/browser/notification_service.h"
+#include "extensions/browser/notification_types.h"
+#include "extensions/browser/updater/update_service.h"
+#include "extensions/browser/updater/update_service_factory.h"
+#include "native_mate/dictionary.h"
+
+
+void ComponentsUI::OnDemandUpdate(
+    component_updater::ComponentUpdateService* cus,
+    const std::string& component_id) {
+  cus->GetOnDemandUpdater().OnDemandUpdate(component_id);
+}
+
+bool
+ComponentsUI::GetComponentDetails(const std::string& id,
+                                  update_client::CrxUpdateItem* item) const {
+  return cus_->GetComponentDetails(id, item);
+}
+
+namespace atom {
+
+namespace api {
+
+UpdateClient::UpdateClient(v8::Isolate* isolate,
+                 content::BrowserContext* browser_context)
+      : browser_context_(browser_context) {
+  cus_ = component_updater::ComponentUpdateServiceFactory(
```

Comment:
it's only used for the default session, even know it can be used for others as well (there would be no point).  Is there a way to get the default session browser context as a singleton or something?